api/utils_api.go

```diff
@@ -5,77 +5,112 @@ import (
 	"log"
 
 	"github.com/gorilla/websocket"
-	"github.com/saeidalz13/battleship-backend/models"
+	cerr "github.com/saeidalz13/battleship-backend/internal/error"
+	md "github.com/saeidalz13/battleship-backend/models"
 )
 
-func CreateGame(s *Server, ws *websocket.Conn) *models.RespCreateGame {
-	newGame := models.NewGame()
-	s.Games[newGame.Uuid] = newGame
+type WsRequestHandler interface {
+	CreateGame() (*md.Message, error)
+	ManageReadyPlayer() (*md.Message, *md.Game, error)
+	JoinPlayerToGame() (*md.Game, md.Message, error)
+}
+
+type WsRequest struct {
+	Server  *Server
+	Ws      *websocket.Conn
+	Payload []byte
+}
```

Comment:
Why do we need this?

Reply:
I'm mostly thinking of it as code organization and it tells us exactly how we're dealing with our requests. In a sense that every request handling is now structured. Also we can kinda keep track of all of our handler functions. If we have a request, we define how it should be dealt with by the interface.

That being said, it's not necessary and not particularly solving any problem... and now I'm not sure if it's adding extra overhead but that should be negligible.

Reply:
Makes sense. Let's keep it then